src/test/resources/features/Users.feature

```diff
@@ -1,119 +1,84 @@
-    And the authorized client forms a "GET" request
-    And the request has the following path parameters:
-      | parameter | value               |
-      | username  | someone@institution |
-    When the authorized client sends the request to read the user
-    Then a user description is returned
+    When the AuthorizedClient sends a request to read the ExistingUser from the Database
+    Then the response object contains the UserDescription of the ExistingUser
 
   Scenario: Authorized client gets a non-existing user
-    Given that a user entry with the username "someone@institution" does not exist in the database
-    And the authorized client forms a "GET" request
-    And the request has the following path parameters:
-      | parameter | value               |
-      | username  | someone@institution |
-    When the authorized client sends the request to read the user
+    When the AuthorizedClient sends a request to read the NonExistingUser from the Database
     Then a NotFound message is returned
 
   Scenario:  Authorized client updates existing user
-    Given that a user entry with the username "someone@institution" exists in the database
-    And the user entry contains a list of roles with the following role-names
-      | role-name |
-      | RoleA     |
-    And the authorized client forms a "PUT" request
-    And the request has the following path parameters:
-      | parameter | value               |
-      | username  | someone@institution |
-    And the request contains a JSON body with following key-value pairs
-      | key         | value               |
-      | username    | someone@institution |
-      | institution | institution         |
-      | type        | User                |
-    And the request body also contains a list of roles with the following role-names
-      | role-name |
-      | RoleB     |
-    When the authorized client sends the request to update the user
-    Then the user entry is updated asynchronously
-    And a Location header with the updated user URI is included in the response
-
-  Scenario:Authorized client attempts to update non-existing user
-    Given that a user entry with the username "someone@institution" does not exist in the database
-    And the authorized client forms a "PUT" request
-    And the request has the following path parameters:
-      | parameter | value               |
-      | username  | someone@institution |
-    And the request contains a JSON body with following key-value pairs
-      | key         | value               |
-      | username    | someone@institution |
-      | institution | institution         |
-      | type        | User                |
-    And the request body also contains a list of roles with the following role-names
-      | role-name |
-      | RoleB     |
-    When the authorized client sends the request to update the user
+    When the AuthorizedClient requests to update the ExistingUser and sets the following roles:
+      | roles |
```

Comment:
And the request body has the following roles

Reply:
This would create a misunderdansting before the API expects a whole User object and not only the roles.